camligorkem/assignment4/CountIslands.java

```diff
@@ -0,0 +1,126 @@
+import java.awt.Point;
+import java.util.HashSet;
+import java.util.Stack;
+import static java.lang.Math.abs;
+
+public class CountIslands {
```

Comment:
While it's great that you show that you can make two different implementation of the solution, I think you should choose only one of them. This is also an exercise in making a decision which solution is better and coming up with reasons. Compare the two algorithms, their performance and clarity of code and choose which one you prefer.